integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py

```diff
@@ -0,0 +1,277 @@
+            #   Triggered via Apache Airflow by task '[a-zA-Z0-9_]+' in the [a-zA-Z-0-9_]+ DAG.
+            regex_pattern = (
+                "Triggered via Apache Airflow by task '([a-zA-Z0-9_]+)' "
+                "in the ([a-zA-Z0-9_]+) DAG\\."
+            )
+            m = re.search(regex_pattern, run_reason)
+            task_id = m.group(1)
+            dag_id = m.group(2)
+
+            if not artifacts.get("run_results", None):
+                continue
+
+            processor = DbtCloudArtifactProcessor(
+                producer=_PRODUCER,
+                job_namespace=_DAG_NAMESPACE,
+                skip_errors=False,
+                logger=log,
+                manifest=manifest,
+                run_result=artifacts["run_results"],
+                profile=connection,
+                catalog=catalog,
+            )
+
+            # parent run is an Airflow task triggering DBT Cloud run
+            parent_run_id = self.context["task_uuid"]
+
+            # if parent run id exists, set it as parent run metadata - so that
+            # the DBT job can contain parent information (which is the current task)
+            if parent_run_id is not None:
+                parent_job = ParentRunMetadata(
+                    run_id=parent_run_id,
+                    job_name=job_name
+                    if job_name is not None
+                    else f"{dag_id}.{task_id}",
+                    job_namespace=_DAG_NAMESPACE,
+                )
+                processor.dbt_run_metadata = parent_job
+
+            events = processor.parse().events()
+            client = OpenLineageAdapter.get_or_create_openlineage_client()
+            for event in events:
+                client.emit(event)
+
+    async def get_dbt_artifacts(
+        self, hook, run_id, account_id, step, artifacts, headers, tenant
+    ):
+        async with aiohttp.ClientSession(headers=headers) as session:
+            tasks = {
+                artifact: DbtCloudExtractor.get_job_run_artifact(
+                    hook=hook,
+                    session=session,
+                    tenant=tenant,
+                    run_id=run_id,
+                    path=artifact,
+                    account_id=account_id,
+                    step=step,
+                )
+                for artifact in artifacts
+            }
+            results = await asyncio.gather(*tasks.values())
+        return {filename: result for filename, result in zip(tasks.keys(), results)}
+
+    def run_io_tasks_in_parallel(self, tasks):
+        with ThreadPoolExecutor() as executor:
+            running_tasks = [executor.submit(task) for task in tasks]
+            for running_task in running_tasks:
+                running_task.result()
+
+    def extract_on_complete(self, task_instance) -> Optional[TaskMetadata]:
+        task_meta_data = self.get_task_metadata()
+        operator = self.operator
+        if (
+            operator.__class__.__name__ == "DbtCloudRunJobOperator"
```

Comment:
`operator.task_type`?

Reply:
Should also check for `Async` versions I think

Reply:
Changed and updated checks against async as well.